slam-libraries/viam-cartographer/src/io/file_handler_test.cc

```diff
@@ -0,0 +1,183 @@
+                                     boost::filesystem::unique_path();
+    bool ok = boost::filesystem::create_directory(tmpdir);
+    if (!ok) {
+        throw std::runtime_error("could not create directory: " +
+                                 tmpdir.string());
+    }
+    for (std::string file : files) {
+        boost::filesystem::ofstream ofs(tmpdir / file);
+        ofs.close();
+    }
+    // Remove a file
+    int success = RemoveFile(tmpdir.string() + "/" + files.at(1));
+    BOOST_TEST(success == 1);
+    files.erase(files.begin() + 1);
+    // List the files in the directory and check if the right number of files
+    // and the right files are still in the directory
+    std::vector<std::string> listedFiles =
+        ListSortedFilesInDirectory(tmpdir.string());
+    BOOST_TEST(files.size() == listedFiles.size());
+    for (int i = 0; i < files.size(); i++) {
+        BOOST_TEST(files.at(i).compare(listedFiles.at(i)));
+    }
+    // Close the file and remove the temporary directory and its contents
+    boost::filesystem::remove_all(tmpdir);
+}
+
+BOOST_AUTO_TEST_CASE(TimedPointCloudDataFromPCDBuilder_success) {
+    // Create a mini PCD file and save it in a tmp directory
+    std::string filename = "rplidar_data_2022-01-01T01:00:00.0001Z.pcd";
+    std::string pcd =
+        "VERSION .7\n"
+        "FIELDS x y z rgb\n"
+        "SIZE 4 4 4 4\n"
+        "TYPE F F F I\n"
+        "COUNT 1 1 1 1\n"
+        "WIDTH 3\n"
+        "HEIGHT 1\n"
+        "VIEWPOINT 0 0 0 1 0 0 0\n"
+        "POINTS 3\n"
+        "DATA ascii\n"
+        "-0.001000 0.002000 0.005000 16711938\n"
+        "0.582000 0.012000 0.000000 16711938\n"
+        "0.007000 0.006000 0.001000 16711938\n";
+    // Create a unique path in the temp directory and add the PCD file
+    boost::filesystem::path tmpdir = boost::filesystem::temp_directory_path() /
+                                     boost::filesystem::unique_path();
+    bool ok = boost::filesystem::create_directory(tmpdir);
+    if (!ok) {
+        throw std::runtime_error("could not create directory: " +
+                                 tmpdir.string());
+    }
+    boost::filesystem::ofstream ofs(tmpdir / filename);
+    ofs << pcd;
+    ofs.close();
+    // Read it in and check if the data in the TimedPointCloudData is equivalent
+    // to what we had in the pcd file
+    cartographer::sensor::TimedPointCloudData timed_pcd =
+        TimedPointCloudDataFromPCDBuilder(tmpdir.string() + "/" + filename, 0);
+
+    BOOST_TEST(timed_pcd.ranges.size() == 3);
+    cartographer::sensor::TimedRangefinderPoint point_1 =
+        timed_pcd.ranges.at(0);
+    cartographer::sensor::TimedRangefinderPoint point_2 =
+        timed_pcd.ranges.at(1);
+    cartographer::sensor::TimedRangefinderPoint point_3 =
+        timed_pcd.ranges.at(2);
+
+    auto tolerance = boost::test_tools::tolerance(0.00001);
+    BOOST_TEST(point_1.position(0, 0) == -0.001, tolerance);
```

Comment:
[opt] similar to above comment, I would make a list of the three pc values that then would be placed in the pcd file and could be referenced below so that if the test changes in the future it may only need to be edited in one place.